indra/llxml/llxmlnode.cpp

```diff
@@ -650,34 +650,106 @@ bool LLXMLNode::updateNode(
 	return true;
 }
 
+static std::map<std::string, LLXMLNodePtr> sXMLCache;
+static LLSharedMutex sXMLCacheMutex;
```

Comment:
fwiw you could consider `LockStatic`, although I see that lockstatic.h lacks a usage example in its comments.

Also `LockStatic` should be extended to support the shared/exclusive lock distinction you use below.